diff -up xorg-server-1.20.10/hw/xwayland/xwayland.c.47~ xorg-server-1.20.10/hw/xwayland/xwayland.c
--- xorg-server-1.20.10/hw/xwayland/xwayland.c.47~	2020-12-01 23:44:55.115023628 +0000
+++ xorg-server-1.20.10/hw/xwayland/xwayland.c	2020-12-01 23:45:56.348511824 +0000
@@ -738,6 +738,7 @@ ensure_surface_for_window(WindowPtr wind
     struct xwl_screen *xwl_screen;
     struct xwl_window *xwl_window;
     struct wl_region *region;
+    WindowPtr toplevel;
 
     if (xwl_window_from_window(window))
         return TRUE;
@@ -808,7 +809,14 @@ ensure_surface_for_window(WindowPtr wind
 
     xwl_window_init_allow_commits(xwl_window);
 
-    if (!window_is_wm_window(window)) {
+    /* When a new window-manager window is realized, then the randr emulation
+     * props may have not been set on the managed client window yet.
+     */
+    if (window_is_wm_window(window)) {
+        toplevel = window_get_client_toplevel(window);
+        if (toplevel)
+            xwl_output_set_window_randr_emu_props(xwl_screen, toplevel);
+    } else {
         /* CSD or O-R toplevel window, check viewport on creation */
         xwl_window_check_resolution_change_emulation(xwl_window);
     }
@@ -857,8 +865,6 @@ xwl_realize_window(WindowPtr window)
             return FALSE;
     }
 
-    xwl_output_set_window_randr_emu_props(xwl_screen, window);
-
     return ensure_surface_for_window(window);
 }
 
