diff -up xorg-server-1.20.2/dix/events.c.5002~ xorg-server-1.20.2/dix/events.c
--- xorg-server-1.20.2/dix/events.c.5002~	2018-10-15 17:59:33.000000000 +0200
+++ xorg-server-1.20.2/dix/events.c	2018-10-17 18:18:56.318388289 +0200
@@ -601,6 +601,7 @@ XineramaConfineCursorToWindow(DeviceIntP
         return;
 
     i = PanoramiXNumScreens - 1;
+    assert(i>=0);
 
     RegionCopy(&pSprite->Reg1, &pSprite->windows[i]->borderSize);
     off_x = screenInfo.screens[i]->x;
