diff -Naur xorg-server-1.17.1/glamor/Makefile.am xorg-server-1.17.1.tpg/glamor/Makefile.am
--- xorg-server-1.17.1/glamor/Makefile.am	2014-11-21 03:02:48.000000000 +0000
+++ xorg-server-1.17.1.tpg/glamor/Makefile.am	2015-02-21 00:11:30.671390249 +0000
@@ -1,6 +1,7 @@
 noinst_LTLIBRARIES = libglamor.la libglamor_egl_stubs.la
 
-libglamor_la_LIBADD = $(GLAMOR_LIBS)
+libglamor_la_LDFLAGS = -module -avoid-version $(LD_NO_UNDEFINED_FLAG)
+libglamor_la_LIBADD = $(GLAMOR_LIBS) $(DLOPEN_LIBS)
 
 AM_CFLAGS = $(CWARNFLAGS) $(DIX_CFLAGS) $(GLAMOR_CFLAGS)
 
