diff -Naur xorg-server-1.16.0/hw/xfree86/common/compiler.h xorg-server-1.16.0.tpg/hw/xfree86/common/compiler.h
--- xorg-server-1.16.0/hw/xfree86/common/compiler.h	2014-06-05 05:22:57.000000000 +0000
+++ xorg-server-1.16.0.tpg/hw/xfree86/common/compiler.h	2014-08-28 20:56:12.212839293 +0000
@@ -1356,7 +1356,7 @@
     !defined(__sh__) && !defined(__hppa__) && !defined(__s390__) && \
     !defined(__m32r__) && !defined(__aarch64__) && !defined(__arc__) && \
     !defined(__xtensa__)
-#ifdef GCCUSESGAS
+#ifdef __GNUC__
 
 /*
  * If gcc uses gas rather than the native assembler, the syntax of these
