--- xorg-server-1.17.1/hw/xfree86/drivers/modesetting/Makefile.am.omv~	2015-02-15 07:56:55.931914753 +0100
+++ xorg-server-1.17.1/hw/xfree86/drivers/modesetting/Makefile.am	2015-02-15 07:57:27.589191730 +0100
@@ -39,7 +39,7 @@ AM_CPPFLAGS = \
 
 modesetting_drv_la_LTLIBRARIES = modesetting_drv.la
 modesetting_drv_la_LDFLAGS = -module -avoid-version
-modesetting_drv_la_LIBADD = $(UDEV_LIBS) $(DRM_LIBS) $(GLAMOR_LIBS) $(top_builddir)/glamor/libglamor.la $(top_builddir)/glamor/libglamor_egl_stubs.la
+modesetting_drv_la_LIBADD = $(UDEV_LIBS) $(DRM_LIBS) $(GLAMOR_LIBS) $(MIEXT_SHADOW_LIB) $(top_builddir)/glamor/libglamor.la $(top_builddir)/glamor/libglamor_egl_stubs.la
 modesetting_drv_ladir = @moduledir@/drivers
 
 modesetting_drv_la_SOURCES = \
