diff -up xorg-server-1.20.4/hw/xfree86/ddc/interpret_edid.c.omv~ xorg-server-1.20.4/hw/xfree86/ddc/interpret_edid.c
--- xorg-server-1.20.4/hw/xfree86/ddc/interpret_edid.c.omv~	2019-02-26 23:39:38.887337913 +0100
+++ xorg-server-1.20.4/hw/xfree86/ddc/interpret_edid.c	2019-02-26 23:39:44.195425899 +0100
@@ -225,7 +225,7 @@ handle_cea_detail_block(Uchar * ext, xf8
                         handle_detailed_fn fn, void *data)
 {
     int i;
-    struct detailed_monitor_section det_mon[CEA_EXT_DET_TIMING_NUM];
+    struct detailed_monitor_section det_mon[CEA_EXT_DET_TIMING_NUM+1];
     int det_mon_num;
 
     det_mon_num = get_cea_detail_timing(ext, mon, det_mon);
